pkg/domain/infra/abi/containers.go

```diff
@@ -913,7 +913,7 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 	} else {
 		report.ExitCode = int(ecode)
 	}
-	if opts.Rm {
+	if opts.Rm && !ctr.ShouldRestart(ctx) {
```

Comment:
We also need to fix the cleanup process to not remove

Reply:
I did but somehow it got lost from the patch.  It is back now.